django_mysql/models/functions.py

```diff
@@ -260,6 +260,31 @@ def __init__(self, expression, path=None, **extra):
         super(JSONLength, self).__init__(*exprs, output_field=output_field)
 
 
+class JSONSet(Func):
+    function = 'JSON_SET'
+
+    def __init__(self, expression, pairs):
+        from django_mysql.models.fields import JSONField
+
+        if isinstance(pairs, dict) and len(pairs.keys()) == 0:
```

Comment:
1. Remove the `isinstance` - firstly, the code currently doesn't block non-dicts, and secondly, we can just allow duck-typing of dict-like objects, no need for strict typing
2. Use `len(pairs)`, no need to extract `keys`